### ISSUE TITLE
Parsing dates in golden dataset

### ISSUE THREAD

**Issue by jchiu0** [dgraph-io/dgraph#352]
When I stick in a printf in UnmarshalText for Date, I got some unexpected output. This is output when I run the loader (for now) on the golden dataset that is recently generated.

The printf:
```
// UnmarshalText unmarshals the data from a text format.
func (v *Date) UnmarshalText(text []byte) error {
	x.Printf("~~~UnmarshalDateText: %s", string(text))
	val, err := time.Parse(dateFormat, string(text))
	if err != nil {
		return err
	}
	return v.fromTime(val)
}
```

The output:
```
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 2010-03-12
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 1982-02-18
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 1978-08-02
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 1964-03-30
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 1951@@http://www.w3.org/2001/XMLSchema#gYear
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 1954@@http://www.w3.org/2001/XMLSchema#gYear
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 2002-11-25
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 1999@@http://www.w3.org/2001/XMLSchema#gYear
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 1921-12-25
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 2005@@http://www.w3.org/2001/XMLSchema#gYear
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 2001@@http://www.w3.org/2001/XMLSchema#gYear
2016/11/30 21:36:32 date.go:85: ~~~UnmarshalDateText: 2004@@http://www.w3.org/2001/XMLSchema#gYear
```

Here is the schema:
```
scalar (
	type.object.name.en: string @index
	film.film.initial_release_date: date @index
)
```

Should the parsing be smarter? Or there's something wrong with the schema? Or it's just something that we will do later?

**Comment by ashwin95r** [comment]
Currently we support : 
```
"http://www.w3.org/2001/XMLSchema#string":   types.StringID,
  "http://www.w3.org/2001/XMLSchema#dateTime": types.DateTimeID,
  "http://www.w3.org/2001/XMLSchema#date":     types.DateID,
  "http://www.w3.org/2001/XMLSchema#int":      types.Int32ID,
  "http://www.w3.org/2001/XMLSchema#boolean":  types.BoolID,
  "http://www.w3.org/2001/XMLSchema#double":   types.FloatID,
  "http://www.w3.org/2001/XMLSchema#float":    types.FloatID,
```
We need to add the rest of the types to the rdf/parse.go and map them to corresponding internal types. And #gYear would map to date type

**Comment by ashwin95r** [comment]
Fixed with b0fe7c51e0c124c12d6294aa18c4190f9f58d919.
